File: src/data/volume_type.rs
use crate::data::icon_type::IconType;

#[derive(Clone, PartialEq)]
pub enum VolumeType {
    Music,
    Rain,
}

impl VolumeType {
    pub fn color(&self) -> &'static str {
        match self {
            VolumeType::Music => "#8959d6",
            VolumeType::Rain => "#1ab3dd",
        }
    }
    pub fn icon(&self) -> IconType {
        match self {
            VolumeType::Music => IconType::Music,
            VolumeType::Rain => IconType::Rain,
        }
    }
}

File: shared/src/commonMain/kotlin/com/example/kncr/data/VolumeType.kt
package com.example.kncr.data

import androidx.compose.ui.graphics.Color

enum class VolumeType {
    Music,
    Rain;

    val color: Color
        get() = when (this) {
            Music -> Color(0xFF8959D6) // #8959d6
            Rain -> Color(0xFF1AB3DD) // #1ab3dd
        }

    val icon: IconType
        get() = when (this) {
            Music -> IconType.Music
            Rain -> IconType.Rain
        }
}
